version: '3'

services:
  gunicorn:
    build:
      context: .
      dockerfile: ./production_Dockerfile
    image: gunicorn:20.1.0
    container_name: nosta_gunicorn
    depends_on:
      - db
    expose:
      - "8000"
    ports:
      - 8000:8000
    networks:
      - nosta
      - prasic
    environment:
      - DJANGO_SETTINGS_MODULE=nosta.settings.production
    volumes:
      - .:/usr/src/app/
      - gunicorn:/var/run/gunicorn
    stdin_open: true
    tty: true
    labels:
      - traefik.enable=true
      - traefik.http.routers.nosta_gunicorn.rule=Host(`nosta.${DOMAIN}`)
      - traefik.http.routers.nosta_gunicorn.entrypoints=https
      - traefik.http.routers.nosta_gunicorn.tls=true
      - traefik.http.services.nosta_gunicorn.loadbalancer.server.port=8000
      - traefik.http.routers.nosta_gunicorn.tls.certresolver=${CERT_RESOLVER}

  db:
    image: mariadb:10.6.4
    restart: always
    container_name: nosta_db
    networks:
      - nosta
    environment:
      MARIADB_ROOT_PASSWORD: nosta_db_root_password
    volumes:
      - ./mysql:/var/lib/mysql

volumes:
  gunicorn:
    driver: local

networks:
  prasic:
    external: true
  nosta:
    external: true
